imap_processing/ialirt/l0/process_swe.py

```diff
@@ -0,0 +1,246 @@
+"""Functions to support I-ALiRT SWE packet parsing."""
+
+import logging
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from numpy.typing import NDArray
+
+from imap_processing.ialirt.utils.grouping import find_groups
+from imap_processing.swe.l1a.swe_science import decompressed_counts
+from imap_processing.swe.l1b.swe_l1b_science import (
+    deadtime_correction,
+    read_in_flight_cal_data,
+)
+from imap_processing.swe.utils.swe_constants import (
+    ESA_VOLTAGE_ROW_INDEX_DICT,
+    GEOMETRIC_FACTORS,
+)
+from imap_processing.swe.utils.swe_utils import combine_acquisition_time
+
+logger = logging.getLogger(__name__)
+
+
+def decompress_counts(raw_counts: NDArray) -> NDArray:
+    """
+    Perform decompression of raw counts using a predefined decompression table.
+
+    Parameters
+    ----------
+    raw_counts : np.ndarray
+        Array of raw compressed counts with shape (n_time, n_cem, n_energy_step).
+
+    Returns
+    -------
+    counts : np.ndarray
+        Array of decompressed counts with the same shape as raw_counts.
+    """
+    decompression_table = np.array([decompressed_counts(i) for i in range(256)])
+
+    # Decompress using the precomputed table
+    counts = decompression_table[raw_counts]
+
+    return counts
+
+
+def phi_to_bin(phi_values: NDArray) -> NDArray:
+    """
+    Convert phi values to corresponding bin indices.
+
+    Parameters
+    ----------
+    phi_values : NDArray
+        Array of phi values.
+
+    Returns
+    -------
+    bin_indices : NDArray
+        Array of bin indices.
+    """
+    # Ensure it wraps correctly within 0-29 bins
+    return ((phi_values - 12) // 12) % 30
+
+
+def prepare_raw_counts(grouped: xr.Dataset, cem_number: int = 7) -> NDArray:
+    """
+    Reformat raw counts into a 3D array binned by phi.
+
+    Parameters
+    ----------
+    grouped : xr.Dataset
+        Dataset containing grouped i-ALiRT packet data for 1 minute.
+    cem_number : int
+        Number of CEMs (default 7).
+
+    Returns
+    -------
+    raw_counts : NDArray
+        Raw counts with shape (8, 7, 30).
```

Comment:
can you change these to be what they are? FYI, `decompress_counts()` function thinks last one is `n_energy_steps`. Can you double check that?
```suggestion
        Raw counts with shape (n_time, n_cem, n_phi).
```

Reply:
I found a typo in decompress_counts(). The raw_counts have dimensions (n_energy_step, n_cem, n_time).

I think this shouldn't matter here, though, I think? Take a look just to make certain.

decompression_table = np.array([decompressed_counts(i) for i in range(256)])

# Decompress using the precomputed table
counts = decompression_table[raw_counts]